test_framework/scripts/longhorn-setup.sh

```diff
@@ -399,16 +398,37 @@ main(){
       get_rancher_api_key
       install_longhorn_by_rancher
     fi
-    run_longhorn_tests ${WORKSPACE}
+    run_longhorn_tests
   elif [[ "${LONGHORN_UPGRADE_TEST}" == true || "${LONGHORN_UPGRADE_TEST}" == True ]]; then
     generate_longhorn_yaml_manifest "${TF_VAR_tf_workspace}"
     install_longhorn_stable
-    run_longhorn_upgrade_test ${WORKSPACE}
-    run_longhorn_tests ${WORKSPACE}
+    LONGHORN_UPGRADE_TYPE="from_stable"
+    LONGHORN_UPGRADE_TEST_POD_NAME="longhorn-test-upgrade-from-stable"
+    if [[ -n "${LONGHORN_TRANSIENT_VERSION}" ]]; then
+      UPGRADE_LH_REPO_URL="${LONGHORN_REPO_URI}"
+      UPGRADE_LH_REPO_BRANCH="${LONGHORN_TRANSIENT_VERSION}"
+      UPGRADE_LH_MANAGER_IMAGE=""
+      UPGRADE_LH_ENGINE_IMAGE="longhornio/longhorn-engine:${LONGHORN_TRANSIENT_VERSION}"
+      UPGRADE_LH_INSTANCE_MANAGER_IMAGE=""
+      UPGRADE_LH_SHARE_MANAGER_IMAGE=""
+      UPGRADE_LH_BACKING_IMAGE_MANAGER_IMAGE=""
```

Comment:
Do we need to fill in the empty value here?
```bash
      UPGRADE_LH_MANAGER_IMAGE=""
      UPGRADE_LH_INSTANCE_MANAGER_IMAGE=""
      UPGRADE_LH_SHARE_MANAGER_IMAGE=""
      UPGRADE_LH_BACKING_IMAGE_MANAGER_IMAGE=""
```

Reply:
IMO keeping only `${LONGHORN_TRANSIENT_VERSION}` specific variable assignments is easier to read. But that is just my 2 cents.

Reply:
refined!